src/Symfony/Component/Finder/Iterator/MultiplePcreFilterIterator.php

```diff
@@ -65,4 +59,28 @@ protected function isRegex($str)
      * @return string regexp corresponding to a given string
      */
     abstract protected function toRegex($str);
+
+    private function buildRegexps($patterns)
+    {
+        $rxs = array();
+        $regexps = array();
+
+        foreach ($patterns as $pattern) {
+            $regex = Regex::create($this->toRegex($pattern))->render();
+
+            if (preg_match('/\W([imsxUXJ]*)$/', $regex, $match)) {
+                $rxs[] = '(?'.$match[1].':'.substr($regex, 1, -strlen($match[0])).')';
+            } else {
+                $regexps[] = $regex;
+            }
+        }
+        if (1 < count($rxs)) {
+            $regexps[] = Regex::BOUNDARY.implode('|', $rxs).Regex::BOUNDARY;
+        } elseif ($rxs) {
+            $rxs = explode(':', substr($rxs[0], 2, -1), 2);
+            $regexps[] = Regex::BOUNDARY.$rxs[1].Regex::BOUNDARY.$rxs[0];
+        }
+
+        return $regexps;
+    }
```

Comment:
this logic will break regexes using complex features, for instance ``(?:^|/)(\w)\w+\1(?:/|$)`` to match directories starting and ending with the same char in the PathFilter.
combining regexes together is impossible if you don't have any knowledge about these regexes.

I suggest you to revert this change for now, to make the PR mergeable.

I have an idea to optimize the existing filters extending this class though, by using some knowledge about the way regexes are built currently (because the user input may not be a regex but something we are turning into a regex, and so we can optimize these cases). I will work on it this evening if I can.